### PR TITLE
Implement free-fall detection.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/lis3dh.toit [hunk 1]
```diff
@@ -99,10 +125,66 @@ class Lis3dh:
...
     reg_.write_u8 CTRL_REG1_ ctrl1
+    reg_.write_u8 CTRL_REG3_ ctrl3
     reg_.write_u8 CTRL_REG4_ ctrl4
+    reg_.write_u8 CTRL_REG5_ ctrl5
+
+    reg_.write_u8 INT1_THS_ int1_threshold
+    reg_.write_u8 INT1_DURATION_ int1_duration
+    reg_.write_u8 INT1_CFG_ int1_cfg
 
     sleep --ms=7 // Wait 7 Mhz to give the sensor time to wake up.
```
Comment (erikcorry) [inline src/lis3dh.toit]: Strange comment.

File: src/lis3dh.toit [hunk 2]
```diff
@@ -99,10 +125,66 @@ class Lis3dh:
...
+      if rate == RATE_1HZ: int1_duration = free_fall_duration.in_s
+      else if rate == RATE_10HZ: int1_duration = free_fall_duration.in_ms / 100
+      else if rate == RATE_25HZ: int1_duration = free_fall_duration.in_ms / 40
+      else if rate == RATE_50HZ: int1_duration = free_fall_duration.in_ms / 20
+      else if rate == RATE_100HZ: int1_duration = free_fall_duration.in_ms / 10
+      else if rate == RATE_200HZ: int1_duration = free_fall_duration.in_ms / 5
+      else if rate == RATE_400HZ: int1_duration = (free_fall_duration * 4).in_ms / 10
+      else: unreachable
+
+      if int1_duration == 0: throw "Invalid rate/duration combination."
```
Comment (erikcorry) [inline src/lis3dh.toit]: We don't normally end exceptions with a '.'.

**Review by erikcorry — APPROVED** [review]

File: src/lis3dh.toit (hunk 2 above)
Comment (floitsch) [reply src/lis3dh.toit]: done

File: src/lis3dh.toit (hunk 1 above)
Comment (floitsch) [reply src/lis3dh.toit]: done.

**Pull request merged**